hil-test/tests/gpio.rs

```diff
@@ -279,4 +281,28 @@ mod tests {
         assert_eq!(io2.is_low(), true);
         assert_eq!(io3.is_set_low(), true);
     }
+
+    // Tests `AnyPin` and GPIO that can be used as a `touch` pin
+    // https://github.com/esp-rs/esp-hal/issues/1943
+    #[test]
+    async fn test_async_gpio(ctx: Context<'static>) {
+        test_any_pin(AnyPin::new(ctx.io4)).await;
```

Comment:
I don't really understand why you don't place the actual test logic into this function. I don't think we want to reuse it, there is no reason that I can see to introduce this sort of indirection.

Can you please add a second pin to this test? We already connect GPIO 2 and 3, sp we can test inputs, not just reading back the output state.

Thirdly, I think if you want to wait in this test, that would be better between writing a pin and reading the input. I don't see what we win by waiting after asserting, but before writing the next pin level.

Also, the last wait is unnecessary :)